plugins/inputs/nightscout/nightscout.go

```diff
@@ -0,0 +1,198 @@
+// Package nightscout will gather blood gloucose data from nightscout sites using the site's api.
+package nightscout
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// sampleConfig will provide an example and documentation for configuring the plugin.
+var sampleConfig = `
+  ## This is a list of nightscout sites to pull data from.
+  ##  example: servers = ["https://my.nightscout.site/api/v1/entries"]
+  servers = []
+  ## The person is the owner of the nightscout site. The order of the list of owners must match the order of the
+  ## list of servers, for example the first server should match the first owner.
+  # owners = ""
+
+  ## The secret is the SHA-1 hash of the api secret used to access the site.
+  secret = ""
+
+  ## The count is the number of results to pull at a time. Unless you have a specific need, this should remain as 1.
+  # count = "1"
+
+  ## The token is the unique token that matches a role within your nightscout site(s).
+  token = ""
+`
+
+// BGData will hold all stats fetched from the database of a nightscout site.
+type BGData struct {
+	Type       string `json:"type"`
+	DateString string `json:"dateString"`
+	Date       int    `json:"date"`
+	SGV        int    `json:"sgv"`
+	Direction  string `json:"direction"`
+	Noise      int    `json:"noise"`
+	Filtered   int    `json:"filtered"`
+	Unfiltered int    `json:"unfiltered"`
+	RSSI       int    `json:"rssi"`
+}
+
+// Nightscout holds configuration and query parameters.
+type Nightscout struct {
```

Comment:
Maybe instead of a flat structure with some lists in it, it would make more sense to have a list of dictionaries? So something like (YAML representation, can try to put it into something else if you prefer):
```yaml
nightscout:
  count: 1
  servers:
    - address: https://server1
      owner: owner1
      token: token1
    - address: https://server2
      owner: owner2
      secret: secret2
```

Reply:
What I'd suggest for this is to use multiple plugin instances:
```toml
[[inputs.nightscout]]
  servers = ["https://server1"]
  token: "token1"

[[inputs.nightscout]]
  servers = ["https://server2"]
  token: "token2"
```

It may even be worth considering having only a single server url plugin instead of a list.  We should think about the typical pattern for connecting, does one usually connect to a single server or to a cluster?

Reply:
@danielnelson I was unaware you could do multiple plugin instances. I am 100% for that and will refactor with that in mind. Thanks for the tip!

Reply:
I will also add more documentation in the readme to discuss use cases for both single person and multiple person uses.

For a single person, though you connect to a single server, there is no real concept of clustering for theses site. You could do something with a load balancer and share the MongoDB backend but I have not heard of anyone needing that kind of up, it may well exist though.

Reply:
Made the changes, to include multiple plugins. I left in the slice for ranging through several servers per person in case they have something like that implemented.